docs/designs/2730-visibility.md

```diff
@@ -0,0 +1,391 @@
+  failure, should be captured in the bootstrap log
+* To prevent hidden leaked filesystem space consumption, the bootstrap log
+  should be deleted as part of the `management-cluster delete` or `cluster
+  delete` operations
+
+#### FR4 - Debug capture on failed cluster deployment
+
+During deployment, we capture the kubectl configuration and other details about
+the bootstrap kind cluster and the cluster that is actively being deployed. We
+use some of this to help guide the user today:
+
+```txt
+Failure while deploying management cluster, Here are some steps to investigate the cause:
+Debug:
+kubectl get po,deploy,cluster,kubeadmcontrolplane,machine,machinedeployment -A --kubeconfig /home/test/.kube-tkg/tmp/config_6SDG31q1
+kubectl logs deployment.apps/ -n manager --kubeconfig /home/test/.kube-tkg/tmp/config_6SDG31q1
+```
+
+Since we have this information, we should actually perform some of the
+debugging steps for the end user.
+
+As a base requirement for this proposal, debug commands like getting pod and
+deployment status from the cluster should be run, with the output captured to
+the bootstrap log. We should also capture the error logs from the CAPI provider
+manager pod (assuming things have gotten that far) and include that in the
+bootstrap log.
+
+For the purposes of this design proposal, the requirement is only to capture
+this output so it can be used for later troubleshooting. But since we have this
+information, a future effort of building in some parsing and automatic triaging
+to the commands should be considered.
+
+### Implementation Details/Notes/Constraints
+
+* The main target for these changes are `management-cluster` and `cluster`
+  create commands
+* Output will be different between these two as there are different details
+  that are more relevant for one versus the other
+* The same output principles should be considered for things like `delete`
+  commands and any others that perform a multi-step process to accomplish
+
+#### Process Steps
+
+The main change in output will be the selection of current log messages that are
+emitted. The things we will want to change for this process output will be:
+
+* Print high level steps
+  * Should lead with emoji if the console supports it
+  * High level grouping of the current set of low level steps being performed
+    (e.g. top level grouping of something like "Setting up core package repository"
+    only including the output of the specific URL used for that repo)
+* Limited output for each grouping
+  * Raise log level of existing trace messages so they only print out if the
+    verbosity level is set high enough
+  * Lead with indentation so output is visually obvious as something underneath
+    the top level group
+  * Review the messages that are printed by default so they are very clear and
+    easy for someone to understand from the context of where they are printed in
+    the overall output
+
+#### Cluster API Details
```

Comment:
how to you plan to expose these? and/or where do you plan to store these details?

_would be good to put this in the proposal if it's not already called out somewhere_

Reply:
Added paragraph clearly stating these are things to capture to the debug log.